src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -111,6 +114,11 @@ static <T> Blob constructBlobAndMetadata(
           compressedChunkData = serializedChunk.chunkData.toByteArray();
           chunkLength = compressedChunkData.length;
           compressedChunkDataSize = chunkLength;
+
+          if (internalParameterProvider.setIcebergSpecificFieldsInEp()) {
+            metadataSize = getExtendedMetadataSize(compressedChunkData);
```

Comment:
method name is adding to confusion, lets call it getParquetFooterSize. Also put it in a utils class and not in BlobBuilder.

Reply:
Renamed to `getParquetFooterSize` and moved to `BlobBuilder`.